### ISSUE TITLE
Comments break multi-line macros

### ISSUE THREAD

**Issue by karalabe** [klauspost/asmfmt#9]
There's a small issue with formatting multiline macros. If the macro contains a comment, then the end-of-line marker should actually be places in front of the comment, not after, as otherwise it will be treated as part of the comment and terminates the macro:

Input:
```
#define MACRO \
  MOVQ AB BX \
  \ // Some comment \
  MOVQ BX AX
```

Current output:
```
#define MACRO \
	MOVQ AB BX          \
	           // Some comment \\
	MOVQ BX AX
```

Expected output:
```
#define MACRO \
	MOVQ AB BX          \
	\ // Some comment   \
	MOVQ BX AX
```

The trailing `\` at the end of the comment makes it look cleaner, even though it isn't interpreted in any way.

Another issue probably also caused by the comment-in-the-multi-macro is that the formatting is not final. If I pass the output into another formatting round, it will further change:

```
$ cat a.txt 
#define MACRO \
  MOVQ AB BX \
  \ // Some comment \
  MOVQ BX AX

$ cat a.txt | asmfmt
#define MACRO \
	MOVQ AB BX          \
	           // Some comment \\
	MOVQ BX AX

$ cat a.txt | asmfmt | asmfmt
#define MACRO \
	MOVQ AB BX \

	// Some comment \\
	MOVQ BX AX

$ cat a.txt | asmfmt | asmfmt | asmfmt
#define MACRO \
	MOVQ AB BX \

// Some comment \\
	MOVQ BX AX
```

**Comment by klauspost** [comment]
Ah. I didn't expect line comments to actually work in multiline definitions, so I only worked with block comments.

Thanks for the test case. I will get back with a fix.

**Comment by karalabe** [comment]
In all honestly, block comments probably make much more sense :D

**Comment by karalabe** [comment]
Just checked, Go 1.4.2 doesn't accept line comments within multi-line macros but Go 1.5.2 does.

**Comment by klauspost** [comment]
I don't think we should touch the content of comments,  but it the inconsistency should of course be fixed.

To avoid moving the line comment to far to the right, I think this should be the expected:

```
INPUT: 
#define MACRO \
  MOVQ AB BX \
  \ // Some comment  \
  \ // Another comment
    MOVQ AB BX \
MOVQ BX AX

OUTPUT:
#define MACRO \
	MOVQ AB BX           \
	\ // Some comment  \
	\ // Another comment
	MOVQ AB BX           \
	MOVQ BX AX
```

We keep the backslash, and any spaces inserted before it.

I have a PR ready.

**Comment by karalabe** [comment]
Fixed by #11, cheers :)
